sdk/tables/azure-data-tables/inc/azure/data/tables/tables_sas_builder.hpp

```diff
@@ -0,0 +1,189 @@
+     */
+    Read = 1,
+
+    /**
+     * @brief Indicates that Add is permitted.
+     */
+    Add = 2,
+
+    /**
+     * @brief Indicates that Update is permitted.
+     */
+    Update = 4,
+
+    /**
+     * @brief Indicates that Delete is permitted.
+     */
+    Delete = 8,
+
+    /**
+     * @brief Indicates that all permissions are set.
+     */
+    All = ~0
+  };
+
+  /** @brief Bitwise OR of two values*/
+  inline TablesSasPermissions operator|(TablesSasPermissions lhs, TablesSasPermissions rhs)
+  {
+    using type = std::underlying_type_t<TablesSasPermissions>;
+    return static_cast<TablesSasPermissions>(static_cast<type>(lhs) | static_cast<type>(rhs));
+  }
+
+  /** @brief Bitwise AND of two values*/
+  inline TablesSasPermissions operator&(TablesSasPermissions lhs, TablesSasPermissions rhs)
+  {
+    using type = std::underlying_type_t<TablesSasPermissions>;
+    return static_cast<TablesSasPermissions>(static_cast<type>(lhs) & static_cast<type>(rhs));
+  }
+
+  /**
+   * @brief TableSasBuilder is used to generate a Shared Access Signature (SAS) for an Azure
+   * Storage Tables.
+   */
+  struct TablesSasBuilder final
+  {
+    /**
+     * @brief The optional signed protocol field specifies the protocol permitted for a
+     * request made with the SAS.
+     */
+    SasProtocol Protocol;
+
+    /**
+     * @brief Optionally specify the time at which the shared access signature becomes
+     * valid. This timestamp will be truncated to second.
+     */
+    Azure::Nullable<Azure::DateTime> StartsOn;
+
+    /**
+     * @brief The time at which the shared access signature becomes invalid. This field must
+     * be omitted if it has been specified in an associated stored access policy. This timestamp
+     * will be truncated to second.
+     */
+    Azure::DateTime ExpiresOn;
+
+    /**
+     * @brief Specifies an IP address or a range of IP addresses from which to accept
+     * requests. If the IP address from which the request originates does not match the IP address
+     * or address range specified on the SAS token, the request is not authenticated. When
+     * specifying a range of IP addresses, note that the range is inclusive.
+     */
+    Azure::Nullable<std::string> IPRange;
+
+    /**
+     * @brief An optional unique value up to 64 characters in length that correlates to an
+     * access policy specified for the table.
+     */
+    std::string Identifier;
+
+    /**
+     * @brief The name of the table being made accessible.
+     */
+    std::string TableName;
+
+    /**
+     * @brief The optional start of the partition key values range being made available.
+     */
+    std::string PartitionKeyStart;
+
+    /**
+     * @brief The optional end of the partition key values range being made available.
+     */
+    std::string PartitionKeyEnd;
+
+    /**
+     * @brief The optional start of the row key values range being made available.
+     */
+    std::string RowKeyStart;

```

Comment:
What is the syntax of the `RowKeyStart`, `PartitionKeyStart`, `PartitionKeyEnd`, etc? Are they free form text, or something else?

Reply:
yes tehy are text, the name of the row / partition, other languages threat them the same, as strings.